x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -204,6 +204,24 @@ public SemanticTextFieldMapper build(MapperBuilderContext context) {
             }
             var childContext = context.createChildContext(leafName(), ObjectMapper.Dynamic.FALSE);
             final ObjectMapper inferenceField = inferenceFieldBuilder.apply(childContext);
+
+            if (inferenceId.isConfigured() == false && searchInferenceId.isConfigured() == false) {
+                inferenceId.setValue(DEFAULT_INFERENCE_ID);
+                searchInferenceId.setValue(DEFAULT_SEARCH_INFERENCE_ID);
+            } else if (inferenceId.isConfigured() == false) {
```

Comment:
Why would we throw here, instead of just defaulting if one or neither is specified?

Reply:
The idea here is that the user can't specify only `search_inference_id` and allow `inference_id` to be the default, because there's no guarantee that the user-specified `search_inference_id` is compatible with the default ELSER `inference_id`.

However, there's no guarantee that _any_ `inference_id`/`search_inference_id` combo is compatible until the user tries a query, so maybe this is overly restrictive. Need to think about it some more...

Reply:
For comparison, `search_analyzer` does not need an `analyzer` to be set up. I think this is overly restrictive from a UX perspective, as users may want to use an optimized-for-search version and keep using the defaults.

I don't have a strong preference, but would favour not limiting users here.

Reply:
Good data point @carlosdelest, thank you for surfacing that. I'll rework this to allow `inference_id` to be the default and `search_inference_id` to be user-supplied.